src/handlers/get.ts

```diff
@@ -32,7 +32,17 @@ const getHandler: Handler = async (request, env, ctx, cache) => {
     return responses.BAD_REQUEST;
   }
 
-  const bucketPath = mapUrlPathToBucketPath(requestUrl, env);
+  let bucketPath: string | undefined;
```

Comment:
Imo instead of this crashing, we should handle this within the `parseUrl` method we have? Because it already handles returning Bad Requests.

So IMO, whatever sort of logic that was crashing `mapUrlPathToBucketPath` should IMO be handled on `parseUrl` so that if we have a bad malformed url, that method itself already returns undefined.

Reply:
`decodeURIComponent` is what's throwing. I did consider checking for the error there but that would mean calling `decodeURIComponent` twice unless we want to add it to the return value of `parseUrl` and then pass it in to `mapUrlPathToBucketPath`. However, we call `parseUrl` when handling post requests as well, and we don't need the response from `decodeURIComponent` for cache purges.

An alternative for this would be adding an optional parameter to `parseUrl` to signify that we want to validate the path, ex/ `parseUrl(request.url, /* validatePath: */ true)`